phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -505,8 +519,21 @@ public ConnectionInfo(String zookeeperQuorum, Integer port, String rootNode, Str
             if (null == this.user) {
                 throw new RuntimeException("Acquired null user which should never happen");
             }
+            this.tenantID = tenantID;
         }
-        
+
+        public static String getTenantID(String url) {
```

Comment:
Public function should have javadoc.  Also this clearly returns null Strings.  Consider Optional to enforce callers handle those cases.  While this can be static, why are we treating this differently than getPort, etc and then just returning the tenantID?  Also will make future mocking simpler.